modules/assim.batch/R/autoburnin.R

```diff
@@ -0,0 +1,62 @@
+#' @name getBurnin
+#' @title Calculate burnin value
+#' @description Use `coda::gelman_plot` to calculate first iteration number where Gelman diagnostic never exceeds the specified threshold.
+#' @param jags_out List of MCMC sample matrices or `mcmc.list` object
+#' @param threshold Maximum value of Gelman diagnostic
+#' @param ... Other parameters to `gelman.plot`
+#' @examples
+#'      library(coda)
+#'      data(line)
+#'      burnin <- getBurnin(line, threshold = 1.05)
+#' @author Michael Dietze, Alexey Shiklomanov
+#' @export
+getBurnin <- function(jags_out, threshold = 1.1, ...) {
+    library(coda)
+    if (length(find("logger.info")) == 0) {
+        msg <- message
+    } else {
+        msg <- logger.info
+    }
+    if (!is.mcmc.list(jags_out)) jags_out <- makeMCMCList(jags_out)
+    png("/dev/null")
+    GBR <- try(gelman.plot(jags_out, autoburnin=FALSE, ...))
+    dev.off()
+    if (class(GBR) == "try-error") {
+        msg("Unable to calculate Gelman diagnostic. Assuming no convergence.")
+        return(1)
+    }
+    gbr_shrink <- GBR$shrink[,,2] > threshold
```

Comment:
@ashiklom if you could add `GBR$shrink[,,2,drop=FALSE]` here, this will be compatible with the rest of PDA code :+1: